build_runner/lib/src/entrypoint/run_script.dart

```diff
@@ -100,9 +100,7 @@ class RunCommand extends BuildRunnerCommand {
     var exitCodeCompleter = Completer<int>();
 
     try {
-      var outputMap = options.outputMap ?? {};
-      outputMap.addAll({tempPath: null});
-
+      var outputMap = (options.outputMap ?? {})..addAll({tempPath: '.'});
```

Comment:
I don't think we want `.` for the output map here, not actually sure what that would do? I think this is supposed to be the top level directory to build, and `null` I believe means all directories, which is probably a decent default.

Reply:
Actually this should be able to just provide the same top level directory as the script it wants to run, so something like `p.split(scriptName).first`. There is a question as to whether this should be `p.url.split` or `p.split`.... presumably `p.split` though which will expect a platform specific path.

Reply:
I'm actually a little confused on this. What do you by mean by "top level directory"? For example, if the user says `pbr run bin/main.dart`, would `tempPath` be `bin/`?

Reply:
So `tempPath` would still be `bin/`, but the value at that key (currently `'.'`) would be `bin`. That would allow us to only build the bin directory.

This does mean we would also hoist that directory up to the top level though which could break some things... that decision was made to mimic what `pub build` used to do (hoist the web dir up to the top level).

Reply:
So, in that case, would `p.dirname(scriptName)` work? Or should I just go ahead with `p.split(scriptName).first`?

Reply:
`p.split(scriptName).first` would be better. If they want to run `bin/something/foo.dart` the top-level directory is `bin/` and that's the granularity we're able to build.